doc/development/guide.rst

```diff
@@ -1,279 +1,45 @@
-automatically to the comment box when you create a new issue.
-
-* When describing the pull request, please include as much detail as possible
-  regarding the changes made/new features added/performance improvements. If including any
-  bug fixes, mention the issue numbers associated with the bugs.
-
-* Once you have submitted the pull request, three things will automatically occur:
-
-  - The **test suite** will automatically run on `Travis CI <https://travis-ci.org/XanaduAI/pennylane>`_
-    to ensure that all tests continue to pass.
-
-  - Once the test suite is finished, a **code coverage report** will be generated on
-    `Codecov <https://codecov.io/gh/XanaduAI/pennylane>`_. This will calculate the percentage
-    of PennyLane covered by the test suite, to ensure that all new code additions
-    are adequately tested.
-
-  - Finally, the **code quality** is calculated by
-    `Codefactor <https://app.codacy.com/app/XanaduAI/pennylane/dashboard>`_,
-    to ensure all new code additions adhere to our code quality standards.
-
-Based on these reports, we may ask you to make small changes to your branch before
-merging the pull request into the master branch. Alternatively, you can also
-`grant us permission to make changes to your pull request branch
-<https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/>`_.
-
-.. _adding_new_templates:
-
-.. include:: adding_templates.rst
+Development guide
+=================
+
+The Development guide contains information regarding how to contribute to the
+PennyLane codebase.
+
+The guides below are aimed towards developers and cover how to install
+PennyLane and its dependencies in development mode, run and add tests, write
+documentation, as well as general best practices and an architectural overview
+of PennyLane, its plugins and devices.
+
+.. devicegalleryitem::
+    :name: Installation
+    :description: Dependencies and installation of PennyLane source code using development mode.
+    :link: guide/installation.html
+
+.. devicegalleryitem::
+    :name: Software tests
+    :description: Installing dependencies and running PennyLane tests and coverage.
+    :link: guide/tests.html
+
+.. devicegalleryitem::
+    :name: Documentation
+    :description: Building and contributing modules and packages to the PennyLane documentation.
+    :link: guide/documentation.html
+
+.. devicegalleryitem::
+    :name: Submitting a pull request
+    :description: Creating and submitting a pull request to the PennyLane repository.
+    :link: guide/pullrequests.html
+
+.. devicegalleryitem::
+    :name: Best practices
+    :description: General best practices for contributing and developing for PennyLane.
+    :link: guide/bestpractices.html
+
+.. devicegalleryitem::
+    :name: Architectural overview
+    :description: Architectural overview of PennyLane, its plugins and devices.
+    :link: guide/architecture.html
```

Comment:
I'd personally not include these until they are added

Reply:
What do you think of simply commenting out these buttons for now, and then whenever the guides are added we could just un-comment them in again?

Reply:
Works fine with me! :)

Reply:
Thanks that's a fair point :) From my perspective, I'd still be more comfortable with having this not appear, should the content not make it into the next release

Reply:
I removed the empty files and commented out the title-cards and the hidden toctree page references. :slightly_smiling_face: I think either way works, but I'm actually leaning towards agreeing with @antalszava that it's better to not add things until they are actually being worked on (in case of things being delayed or changed).